forge-ai/src/main/java/forge/ai/AiController.java

```diff
@@ -1623,8 +1626,15 @@ private SpellAbility chooseSpellAbilityToPlayFromList(final List<SpellAbility> a
             AiPlayDecision aiPlayDecision = AiPlayDecision.CantPlaySa;
             if (useLivingEnd) {
                 if (sa.isCycling() && sa.canCastTiming(player)) {
-                    if (ComputerUtilCost.canPayCost(sa, player, sa.isTrigger()))
-                        aiPlayDecision = AiPlayDecision.WillPlay;
+                    if (ComputerUtilCost.canPayCost(sa, player, sa.isTrigger())) {
```

Comment:
Honestly it makes no sense to me to create this "aggressive decision override" mode and now try to backport all kinds of sanity logic we'd normally benefit from into this single place
E.g. it's only a matter of time until we get a report where the AI cycles itself into a loss because the library was empty
Perhaps the way this mode gets enabled can be tweaked as part of the deck selection or something :/
sorry, I really don't want to sound too harsh and all...

Reply:
I have no idea tbh, though I do agree with the reasoning. Not sure if I'm the best person to do the latter though, as I'm not the author of the override logic and I'm not sure if I fully understand how exactly it can be wired better to avoid this kind of situation ^^;

Reply:
I checked the logic - as far as my possibilities go at the moment (given the available spare time), I can only either patch up the existing implementation and add a TODO for a better implementation to hopefully be coded in the future (either by me or by someone else), or I can revert my code in this branch and leave the implementation broken for both cases (life payment and library size) until someone else can patch it properly or until I can find enough time to dig deep into the question of how to rewire the logic. Please let me know what would be preferable.

Reply:
hmn, let's just leave it alone like this now? 🤔 
It's not like there will be huge negative impact, besides the approach having this general function getting a bit messier with card specific stuff ofc

Reply:
Alright, sounds good, let's keep it as is for now then, and hopefully we can modify it later to have the relevant complication avoided

Reply:
> Honestly it makes no sense to me to create this "aggressive decision override" mode and now try to backport all kinds of sanity logic we'd normally benefit from into this single place E.g. it's only a matter of time until we get a report where the AI cycles itself into a loss because the library was empty Perhaps the way this mode gets enabled can be tweaked as part of the deck selection or something :/ sorry, I really don't want to sound too harsh and all...

I don't see any different play style/variation that use Living End based deck on this stats (Living End Decks): https://mtgdecks.net/prices/living-end

So the decision is forced to pile up graveyard and use cascade to cheat/play Living End (try removing the forced decision and see what the AI will do), The logic is somehow lacking in some aspect checks but I can't introduce more logic without slowing down the AI decision.